src/utils/verification.js

```diff
@@ -0,0 +1,53 @@
+const ver1 = require("../constants/verification");
+
+const nameCheck = (name) => {
+  if (!name) {
+    return ver1.verificationStatus.emptyError;
+  }
+
+  return typeof name === "string"
+    ? ver1.verificationStatus.verification
+    : ver1.verificationStatus.typeError;
+};
+
+const numberCheck = (number) => {
+  if (!number) {
+    return ver1.verificationStatus.emptyError;
+  }
+
+  if (typeof number !== "number") {
+    return ver1.verificationStatus.typeError;
+  }
+
+  return number < 0 || number > 100
+    ? ver1.verificationStatus.rangeError
+    : ver1.verificationStatus.verification;
+};
+
+const verficationCheck = ({ name, ...subjects }) => {
+  if (nameCheck(name) !== ver1.verificationStatus.verification) {
+    return { status: nameCheck(name), value: "name" };
+  }
+
+  return Object.entries(subjects).reduce((acc, [value, subject]) => {
+    const status = numberCheck(subject);
+
+    if (status == ver1.verificationStatus.verification) {
+      if (acc === 0) {
+        return {
+          status,
+          ...(status !== ver1.verificationStatus.verification && { value }),
+        };
+      } else {
+        return acc;
+      }
+    }
+
+    return {
+      status,
+      ...(status !== ver1.verificationStatus.verification && { value }),
+    };
+  }, 0);
```

Comment:
초기값이 왜 0이죠..? 내가 알려준 로직은 어디가고 이렇게 짜심?

Reply:
```suggestion
    if (acc.status && acc.status !== ver1.verificationStatus.verification) {
    	return acc;
    }
    const status = numberCheck(subject);

    return {
      status,
      ...(status !== ver1.verificationStatus.verification && { value }),
    };
  }, {});
```

이렇게 하면 코드 라인수를 훨씬 줄일수 있지 않을까?

Reply:
아 맞네 AND연산자 바로수정하겠음.
수정한거는 알려준 로직 그대로 카피해서 돌려봤는데 잘못입력됐을 때였나? 정답이였을 때였나 둘중 하나만 정상동작 되더라고. 그래서 구조랑 문법 이해할겸 혼자 이것저것 테스트하다가 최종적으로 저래 되버림;;

Reply:
동작이 잘 되면 더 간결하게 이해하기 쉬운 코드가 되도록 고쳐야지 그냥 동작 된다고 해서 다가 아니야. 동작이 되면 더 잘 보기쉬운 코드가 되도록 고민해바